corehq/apps/app_manager/view_helpers.py

```diff
@@ -24,6 +25,8 @@ def app(self):
             # if get_app is mainly used for views,
             # maybe it should be a classmethod of this mixin? todo
             return get_app(self.domain, self.app_id)
+        except Http404 as e:
+            raise e
         except Exception:
```

Comment:
I could see an argument for getting rid of the general ```Exception``` catch, to prevent failing silently.